test/test_autograd.py

```diff
@@ -4185,7 +4185,7 @@ def run_functional_checks(test_case, test_name, name, apply_fn, run_grad_checks,
         test_case.assertEqualTypeString(self_variable, self_variable.grad)
         test_case.assertEqual(self_variable.size(), self_variable.grad.size())
 
-separate_complex_tests = ['log', 'log10', 'log1p', 'log2', 'reciprocal']
+separate_complex_tests = ['log', 'log10', 'log1p', 'log2', 'reciprocal', 'tan']
```

Comment:
Why is this one in the reparate list? It should work for real numbers as well no?

Reply:
the test for floating points uses clamp in the input and clamp does not work for complex